### PR TITLE
[BUG]:  joins with duplicate column names and qualified table expansion

### PR DESCRIPTION
**Pull request by universalmind303**
- closes https://github.com/Eventual-Inc/Daft/issues/3070
- closes https://github.com/Eventual-Inc/Daft/issues/3072
- closes https://github.com/Eventual-Inc/Daft/issues/3073

### CODE REVIEW

File: src/daft-sql/src/planner.rs [hunk 1]
```diff
@@ -480,50 +493,71 @@ impl SQLPlanner {
...
             } else {
-                column_not_found_err!(&column_name, &current_relation.get_name());
+                column_not_found_err!(&ident_str, &current_relation.get_name());
             }
         } else {
             table_not_found_err!(root);
         }
     }
 
     fn select_item_to_expr(&self, item: &SelectItem) -> SQLPlannerResult<Vec<ExprRef>> {
```
Comment (kevinzwang) [inline src/daft-sql/src/planner.rs:533]: To confirm, this converts a select item to an unresolved expression, which should then be resolved in the logical plan?

File: src/daft-sql/src/planner.rs [hunk 2]
```diff
@@ -468,6 +480,7 @@ impl SQLPlanner {
     }
 
     fn plan_compound_identifier(&self, idents: &[Ident]) -> SQLPlannerResult<Vec<ExprRef>> {
```
Comment (kevinzwang) [inline src/daft-sql/src/planner.rs:482]: I know that there are a lot of different things that a compound identifier could represent (duckdb examples: https://duckdb.org/docs/sql/data_types/struct.html#dot-notation-order-of-operations), which ones are we intending to support (in this PR as well as in the future)?

**Review by kevinzwang — COMMENTED** [review]
Generally looks good, but I have some questions about the logic in `src/daft-sql/src/planner.rs` to confirm my understanding. Also down to hop on Zoom to chat about it

File: src/daft-plan/src/logical_plan.rs [hunk 3]
```diff
@@ -264,7 +264,7 @@ impl LogicalPlan {
             [input1, input2] => match self {
                 Self::Source(_) => panic!("Source nodes don't have children, with_new_children() should never be called for Source ops"),
                 Self::Concat(_) => Self::Concat(Concat::try_new(input1.clone(), input2.clone()).unwrap()),
-                Self::Join(Join { left_on, right_on, join_type, join_strategy, .. }) => Self::Join(Join::try_new(input1.clone(), input2.clone(), left_on.clone(), right_on.clone(), *join_type, *join_strategy).unwrap()),
+                Self::Join(Join { left_on, right_on, join_type, join_strategy, .. }) => Self::Join(Join::try_new(input1.clone(), input2.clone(), left_on.clone(), right_on.clone(), *join_type, *join_strategy, None, None).unwrap()),
```
[the remark this answers is not in the document]
Comment (universalmind303) [reply src/daft-plan/src/logical_plan.rs:267]: so the `Join` struct doesn't contain the prefix/suffixes. They are applied during `try_new`. This probably shouldn't be done eagerly like it is, but I didn't want to do a huge refactor on the existing join logic.

File: src/daft-sql/src/planner.rs (hunk 2 above)
Comment (universalmind303) [reply src/daft-sql/src/planner.rs:482]: right now we only support table access `tbl1.column`. struct access is currently only available through bracket notation `column["field"]`. Eventually, I think we should support struct access through dot notation though.

**Pull request merged**